packages/components/src/internal/components/search/SampleFinderSection.tsx

```diff
@@ -165,26 +268,34 @@ export const SampleFinderSamplesImpl: FC<SampleFinderSamplesGridProps & Injected
 const SampleFinderSamplesWithQueryModels = withQueryModels<SampleFinderSamplesGridProps>(SampleFinderSamplesImpl);
 
 const SampleFinderSamples: FC<SampleFinderSamplesProps> = memo(props => {
-    const { cards, ...gridProps } = props;
-
-    const baseFilters = [];
-    // TODO this is not really correct.  Probably there will be a specialized filter type used for these lineage queries.
-    // cards.forEach(card => {
-    //     if (card.filterArray.length) {
-    //         baseFilters.push(...card.filterArray);
-    //     } else  {
-    //         baseFilters.push(card.entityDataType.inputColumnName.replace("First", card.schemaQuery.queryName), null, Filter.Types.NONBLANK);
-    //     }
-    // });
-
-    const queryConfigs = {
-        allSamples: {
-            id: 'allSamples',
-            title: 'All Samples',
-            schemaQuery: SCHEMAS.EXP_TABLES.MATERIALS,
-            requiredColumns: SAMPLE_STATUS_REQUIRED_COLUMNS,
-        },
-    };
+    const { cards, selectionKeyPrefix, user, ...gridProps } = props;
+    const [queryConfigs, setQueryConfigs] = useState<{ [key: string]: QueryConfig }>(undefined);
+    const [errors, setErrors] = useState<string>(undefined);
+
+    useEffect(() => {
+        (async () => {
+            try {
+                setQueryConfigs(undefined);
+                const configs = await getSampleFinderQueryConfigs(user, cards, selectionKeyPrefix);
```

Comment:
it seems a bit odd that getting the query configs each time would need an await. What is the API call happening here? If I'm reading it right, it looks like it is a call to get the sample type names. Is that being done each time we generate query configs? or is that only done once and stashed somewhere in the component state?

Reply:
Ah yes.  Little too eager with the promises here.  Changed so we pass in the sample type names from the page.